libraries/AP_HAL_PX4/UARTDriver.cpp

```diff
@@ -93,19 +87,13 @@ void PX4UARTDriver::begin(uint32_t b, uint16_t rxS, uint16_t txS)
     /*
       allocate the write buffer
      */
-	if (txS != 0 && txS != _writebuf_size) {
+    if (txS != 0 && txS != _writebuf.get_size()) {
         _initialised = false;
         while (_in_timer) {
             hal.scheduler->delay(1);
         }
-		_writebuf_size = txS;
-		if (_writebuf != NULL) {
-			free(_writebuf);
-		}
-		_writebuf = (uint8_t *)malloc(_writebuf_size+16);
-		_writebuf_head = 0;
-		_writebuf_tail = 0;
-	}
+        _writebuf.set_size(txS);
```

Comment:
@tridge can explain since he's the original author. But the only reason I can imagine is to serve as a "guard size" to protect against bugs in the ringbuffer implementation.  So if you write past your size, you will corrupt the packet, but at least you won't corrupt other data in memory (or crash in some cases).  I don't see a need to keep the previous behavior